### PR TITLE
Fixing the operation wait logic in TestClusterAPI

### PR DESCRIPTION
**Pull request by sduskis**

### CODE REVIEW

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 1]
```diff
@@ -157,7 +164,11 @@ private void waitForOperation(BigtableClusterAdminClient client, String operatio
         e.printStackTrace();
       }
       Operation response = client.getOperation(request);
-      if (response.getError() != null) {
+      System.out.println(response);
+      if (response.getDone()) {
+        if (response.hasField(ERROR_DESCRIPTOR)) {
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: Shoudlnt there be a hasError() method?

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 2]
```diff
@@ -157,7 +164,11 @@ private void waitForOperation(BigtableClusterAdminClient client, String operatio
         e.printStackTrace();
       }
       Operation response = client.getOperation(request);
-      if (response.getError() != null) {
+      System.out.println(response);
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: Remove

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 3]
```diff
@@ -157,7 +164,11 @@ private void waitForOperation(BigtableClusterAdminClient client, String operatio
         e.printStackTrace();
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: Also, since you are modifying the logic, We should wait after doing a first check, and correctly propagate the InterruptedException.

**Comment by sduskis** [comment]
@mgarolera  I addressed the two issues: 1) a more idiomatic check for error 2) propagating the interrupt.  PTAL

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 4]
```diff
@@ -147,17 +148,33 @@ private Cluster getCluster(BigtableClusterAdminClient client, String clusterName
     }
   }
 
+  // This is needed to see if there is an error.  The *_NUMBER fields are 1 based, but the
+  // getFields() list is 0 based, so we need to subtract 1 from Operation.ERROR_FIELD_NUMBER 
+  // to get the correct ERROR_DESCRIPTOR
+  final static FieldDescriptor ERROR_DESCRIPTOR =
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: Not needed.

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 5]
```diff
@@ -147,17 +148,33 @@ private Cluster getCluster(BigtableClusterAdminClient client, String clusterName
...
   private void waitForOperation(BigtableClusterAdminClient client, String operationName,
-      int maxSeconds) {
+      int maxSeconds) throws InterruptedException {
     GetOperationRequest request = GetOperationRequest.newBuilder().setName(operationName).build();
     for (int i = 0; i < maxSeconds; i++) {
       try {
         Thread.sleep(1000);
       } catch (InterruptedException e) {
-        e.printStackTrace();
+        Thread.interrupted();
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: why is this here? Now that the method throws Interrupted, you don't need the try statement.

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java [hunk 6]
```diff
@@ -147,17 +148,33 @@ private Cluster getCluster(BigtableClusterAdminClient client, String clusterName
...
+        switch (response.getResultCase()) {
+          case ERROR:
+            throw new RuntimeException("Cluster could not be created: " + response.getError());
+          case RESPONSE:
+            return;
+          case RESULT_NOT_SET:
+            throw new IllegalStateException(
+                "System returned invalid response for Operation check: " + response);
+        }
         return;
```
Comment (mgarolera) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: remove this return?

File: bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java (hunk 5 above)
Comment (sduskis) [inline bigtable-hbase/src/test/java/com/google/cloud/bigtable/hbase/TestClusterAPI.java]: good point

**Comment by sduskis** [comment]
Good catches.  Addressed the 3 issues: 1) remove ERROR_DESCRIPTOR 2) remove InterruptedException try/catch 3) removed the extraneous return.  PTAL

**Comment by mgarolera** [comment]
LGTM
